pkg/landscaper/controllers/execution/controller.go

```diff
@@ -64,6 +64,10 @@ func (c *controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, errHdl(ctx, err)
 	}
 
+	if err := HandleDeployItemGeneration(ctx, c.client, logger, exec); err != nil {
+		return reconcile.Result{}, errHdl(ctx, err)
+	}
+
 	if lsv1alpha1helper.IsCompletedExecutionPhase(exec.Status.Phase) {
 		logger.V(7).Info("execution item is already in completion phase")
```

Comment:
I think we should only check the deployitem generation if we are in a completed phase?
otherwise we are already in a reconciling phase and we would just waste the resources and requests for all deploy items. WDYT?
```suggestion

	if lsv1alpha1helper.IsCompletedExecutionPhase(exec.Status.Phase) {
	    if err := HandleDeployItemGeneration(ctx, c.client, logger, exec); err != nil {
		    return reconcile.Result{}, errHdl(ctx, err)
	    }
		logger.V(7).Info("execution item is already in completion phase")
```